common/lsp/lsp-protocol-enums.h

```diff
@@ -0,0 +1,57 @@
+// Copyright 2021 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef COMMON_LSP_LSP_PROTOCOL_ENUMS_H
+#define COMMON_LSP_LSP_PROTOCOL_ENUMS_H
+
+// Enums are currently not handled yet by jcxxgen, so enumeration are
+// done separately here.
+namespace verible {
+namespace lsp {
+// These are the SymbolKinds defined by the LSP specifcation
+// https://microsoft.github.io/language-server-protocol/specifications/specification-3-17/#symbolKind
+// Interesting is, that not all of them are actually supported by all
+// editors (playing around with Kate (https://kate-editor.org/)). So maybe
+// these editors need to be made understanding.
+enum SymbolKind {
+  File = 1,
+  Module = 2,  // SV module. Kate does not seem to support that ?
+  Namespace = 3,
+  Package = 4,  // SV package
+  Class = 5,    // SV class
+  Method = 6,   // SV class -> method
+  Property = 7,
+  Field = 8,
+  Constructor = 9,
+  Enum = 10,  // SV enum
+  Interface = 11,
+  Function = 12,  // SV function
+  Variable = 13,
+  Constant = 14,
+  String = 15,
+  Number = 16,
+  Boolean = 17,
```

Comment:
How is Boolean not already a Variable? (same general question about the classifications of Variable, Constant, Number)

Reply:
They are all upper-case, so the C++ compiler does not have a name-clash. Or do you mean the general classification ? This is from the Language Server Protocol, so they slapped in all sorts of generic classifications that might be useful in tagging syntax tree nodes.